packages/ozone/src/db/migrations/20250923T000000000Z-scheduled-actions.ts

```diff
@@ -0,0 +1,62 @@
+import { Kysely } from 'kysely'
+
+export async function up(db: Kysely<unknown>): Promise<void> {
+  await db.schema
+    .createTable('scheduled_action')
+    .addColumn('id', 'bigserial', (col) => col.primaryKey())
+    .addColumn('action', 'varchar', (col) => col.notNull())
+    .addColumn('eventData', 'jsonb')
+    .addColumn('did', 'varchar', (col) => col.notNull())
+    .addColumn('executeAt', 'timestamptz')
+    .addColumn('executeAfter', 'timestamptz')
+    .addColumn('executeUntil', 'timestamptz')
+    .addColumn('randomizeExecution', 'boolean', (col) => col.defaultTo(false))
+    .addColumn('createdBy', 'varchar', (col) => col.notNull())
+    .addColumn('createdAt', 'timestamptz', (col) => col.defaultTo('now()'))
+    .addColumn('updatedAt', 'timestamptz', (col) => col.defaultTo('now()'))
+    .addColumn('status', 'varchar', (col) => col.defaultTo('pending'))
+    .addColumn('lastExecutedAt', 'timestamptz')
+    .addColumn('lastFailureReason', 'text')
+    .addColumn('executionEventId', 'bigint')
+    .execute()
+
+  // Unique constraint to prevent multiple pending actions for the same subject
+  await db.schema
+    .createIndex('scheduled_action_unique_pending_subject')
+    .unique()
+    .on('scheduled_action')
+    .columns(['did', 'action', 'status'])
+    .execute()
```

Comment:
I think this may prevent multiple actions of the same type being marked as `'executed'`.  I think the intention is just to avoid multiple `'pending'`, which might be something that either needs to be remodeled or enforcement moved up into the application.

Reply:
right yeah we have the validation in the application layer but want to have something in the db layer too. ended up using this 

```
  await sql`
    CREATE UNIQUE INDEX scheduled_action_unique_pending_subject
    ON scheduled_action (did, action)
    WHERE status = 'pending'
  `.execute(db)
```